File: Python/sort-colors.py
class Solution:
    # @param {integer[]} nums
    # @return {void} Do not return anything, modify nums in-place instead.
    def sortColors(self, nums):
        colors = [0, 0, 0]
        flag = 0
        for n in nums:
            colors[n] += 1
        for i in xrange(len(nums)):
            while colors[flag]<=0:
                flag += 1
            nums[i] = flag
            colors[flag] -= 1

File: C++/sort-colors.cpp
class Solution {
public:
    void sortColors(std::vector<int> &nums) {
        int colors[3] = {0};
        int flag = 0;
        for(int n : nums) colors[n]+=1;
        for(int i=0; i<nums.size(); ++i) {
            while(colors[flag]<=0) ++flag;
            nums[i] = flag;
            --colors[flag];
        }
    }
};

